### PR TITLE
Make unit tests follow the AAA pattern

### LINKED ISSUE #30: AAA Pattern
**Requirement**

We should review all the unit tests to follow the [AAA Pattern](https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics?view=vs-2019).

**Value Proposition**

Using the pattern makes it easier to know what the test is actually testing.
```
//arrange
int value=0;
int expected=1;

//act
var result = plusOne(value);

//assert
Assert(expected,result)
```
**Design Ideas**

Following the AAA Pattern from Microsoft.

### PR DESCRIPTION
**Pull request by ilusi0n**
This PR will mark every unit test follow the AAA pattern.
Feel free to point out improvements or mistakes.

Closes #30

### CODE REVIEW

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs [hunk 1]
```diff
@@ -136,7 +203,15 @@ public void SplitEmptyStrIsSingletonEmptyStr()
         [Category("Unit")]
         public void EqualsIgnoreDigitsTrueWhenBothSidesHaveDifferentDigits()
         {
-            Assert.IsTrue("s000df11asd1hf23las".EqualsIgnoreDigits("sdf33asdhf057la12389s"));
+            //Arrange
+            string value = "s000df11asd1hf23las";
+            string digits = "sdf33asdhf057la12389s";
```
Comment (hctrdev) [inline CsharpExtrasTest/Extensions/StringExtensionTest.cs:208]: Looks like this variable is not used for anything. I think you may have missed replacing the string below with this variable.
The naming of the variables is also not very intuitive as both variables contain digits. Not quite sure what to name them, maybe `valueWithDigits1` and `valueWithDigits2` would make more sense? I'm open to suggestions though.

**Review by hctrdev — CHANGES_REQUESTED** [review]
Thank you for your contribution, this certainly makes it easier to see what is going on in the tests!
I just found one small thing I would change.

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs (hunk 1 above)
Comment (ColmBhandal) [reply CsharpExtrasTest/Extensions/StringExtensionTest.cs:208]: Usually in test cases I would name the variables expected(Something) and actual(Something). Usually the things you are comparing are an expected piece of data vs an actual piece of data that arose from some action.

Comment (hctrdev) [reply CsharpExtrasTest/Extensions/StringExtensionTest.cs:208]: I agree, but in this case both of these variables are test inputs, not actual or expected results. The expected/actual result is a boolean.

Comment (ColmBhandal) [reply CsharpExtrasTest/Extensions/StringExtensionTest.cs:208]: Aaaaah... Thank you for letting me see the light!!

**Comment by ColmBhandal** [comment]
@ilusi0n Do you plan to continue on this PR or should one of us take it?

**Comment by ilusi0n** [comment]
> @ilusi0n Do you plan to continue on this PR or should one of us take it?

Hello. I'm sorry, I had a lot of things to do the last week and didn't still had time to look at it. If you want to continue, it's fine by me. For what I read is not missing that much stuff.

**Comment by ColmBhandal** [comment]
Sure no problem. One of us will take over this PR at some stage.

**Comment by ColmBhandal** [comment]
Closing this PR as I will be opening another one within this repo.

**Pull request closed without merging**